src/routes/transactions/mappers/common/settings-change.mapper.ts

```diff
@@ -47,86 +47,183 @@ export class SettingsChangeMapper {
-          ),
-          Number(
-            this.dataDecodedParamHelper.getValueAtPosition(dataDecoded, 1),
-          ),
-        );
+        return this.handleAddOwnerWithThreshold(dataDecoded);
       case SettingsChangeMapper.REMOVE_OWNER:
-        return new RemoveOwner(
-          new AddressInfo(
-            this.dataDecodedParamHelper.getValueAtPosition(dataDecoded, 1),
-          ),
-          Number(
-            this.dataDecodedParamHelper.getValueAtPosition(dataDecoded, 2),
-          ),
-        );
-
+        return this.handleRemoveOwner(dataDecoded);
       case SettingsChangeMapper.SWAP_OWNER:
-        return new SwapOwner(
-          new AddressInfo(
-            this.dataDecodedParamHelper.getValueAtPosition(dataDecoded, 1),
-          ),
-          new AddressInfo(
-            this.dataDecodedParamHelper.getValueAtPosition(dataDecoded, 2),
-          ),
-        );
-      case SettingsChangeMapper.CHANGE_MASTER_COPY: {
-        const masterCopy = await this.addressInfoHelper.getOrDefault(
-          chainId,
-          safe.address,
-        );
-        return new ChangeMasterCopy(masterCopy);
-      }
-      case SettingsChangeMapper.ENABLE_MODULE: {
-        const module = await this.addressInfoHelper.getOrDefault(
-          chainId,
-          this.dataDecodedParamHelper.getValueAtPosition(dataDecoded, 0),
-        );
-        return new EnableModule(module);
-      }
-      case SettingsChangeMapper.DISABLE_MODULE: {
-        const module = await this.addressInfoHelper.getOrDefault(
-          chainId,
-          this.dataDecodedParamHelper.getValueAtPosition(dataDecoded, 1),
-        );
-        return new DisableModule(module);
-      }
+        return this.handleSwapOwner(dataDecoded);
+      case SettingsChangeMapper.CHANGE_MASTER_COPY:
+        return this.handleChangeMasterCopy(chainId, dataDecoded);
+      case SettingsChangeMapper.ENABLE_MODULE:
+        return this.handleEnableModule(chainId, dataDecoded);
+      case SettingsChangeMapper.DISABLE_MODULE:
+        return this.handleDisableModule(chainId, dataDecoded);
       case SettingsChangeMapper.CHANGE_THRESHOLD:
-        return new ChangeThreshold(
-          this.dataDecodedParamHelper.getValueAtPosition(dataDecoded, 0),
-        );
-      case SettingsChangeMapper.SET_GUARD: {
-        const guardValue = this.dataDecodedParamHelper.getValueAtPosition(
-          dataDecoded,
-          0,
-        );
-        if (guardValue !== NULL_ADDRESS) {
-          const guardAddressInfo = await this.addressInfoHelper.getOrDefault(
-            chainId,
-            guardValue,
-          );
-          return new SetGuard(guardAddressInfo);
-        } else {
-          return new DeleteGuard();
-        }
-      }
+        return this.handleChangeThreshold(dataDecoded);
+      case SettingsChangeMapper.SET_GUARD:
+        return this.handleSetGuard(chainId, dataDecoded);
     }
-    throw new Error('Unknown setting');
+
+    return null;
```

Comment:
This scenario shouldn't be possible, right? I mean, I'm assuming the `method` should be always contained in `SETTINGS_CHANGE_METHODS`, so if that assumption is right, if we reach this code something weird happened. Do you think it's better to return `null` in that case instead of explicitly throwing an error? What about at least including a message in the logs?

Reply:
Yes this should not be possible if we consider that we check that the method is a settings change beforehand.

The current CGW implementation actually returns `null` here but I think that can hide some issues. Will revert this change.

Reply:
But it does return `null` if the method name is known and matches but the arguments don't match.

Reply:
Changed in [7ed250e](https://github.com/5afe/safe-client-gateway-nest/pull/245/commits/7ed250ebe5aefdbd2a26d74cf769acd303da7e77)